build-tools/packages/build-cli/package.json

```diff
@@ -158,6 +158,23 @@
 	"engines": {
 		"node": ">=14.17.0"
 	},
+	"fluidBuild": {
+		"tasks": {
+			"compile": {
+				"dependsOn": [
+					"...",
+					"build:diagrams"
+				],
+				"script": false
```

Comment:
IIRC this will make it so the `compile` script doesn't actually run, right? Only the dependencies declared here. If that's intended, I think we don't need the script at all, just this task, but I suspect it might not be intended.

Reply:
I may have misunderstood you, but yes, this is intentional. fluid-build task definitions can only be invoked by fluid-build, so without the scripts in the individual projects, if you're in one of them like build-cli you have to run `pnpm exec fluid-build . -t compile` to run a task. That's annoying, plus it requires people know and use the fluid-build CLI directly, which just isn't how most people are accustomed to running tasks in the repo. `pnpm compile` is just a shortcut to that.

And the `...` in the dependencies means "all the defaults from the task definition in the fluid-build config."

Reply:
I didn't call it out in the PR description, but I should have -- I'm looking to simplify the script names over time and get rid of the "build:" prefix in a lot of places. I usually start something like that with build-tools as an experiment and if I like the results, I make a proposal to the team to do it more broadly. So that's partly why the compile script was added. It alsp mirrors how the tasks are configured/named in fluid-build. I'll add this to the PR description.

Reply:
Ah, I think I see what's confusing me... even in the root fluid build config, we have tasks whose name matches the script we call to trigger them (`build` being the primary example). So in the fluid build graph we don't want those tasks to execute the corresponding npm script because that's just our "entrypoint" to kick them off, not really something the task is supposed to do as part of its execution. This change makes sense now. Leaves me wondering if we should do something to remove that big of ambiguity, but not for this PR.